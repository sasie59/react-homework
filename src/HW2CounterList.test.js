import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW2CounterList';
test('should HW2CounterList', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should add 1 item', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('should add 3 item', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('should add 5 item', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});



