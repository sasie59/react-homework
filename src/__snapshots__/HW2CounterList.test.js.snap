// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW2CounterList 1`] = `
<div>
  <div>
    <h1>
      HW2CounterList
    </h1>
    <button>
      click me
    </button>
    <ul />
  </div>
</div>
`;

exports[`should add 1 item 1`] = `
<div>
  <div>
    <h1>
      HW2CounterList
    </h1>
    <button>
      click me
    </button>
    <ul>
      <li>
        1
      </li>
    </ul>
  </div>
</div>
`;

exports[`should add 3 item 1`] = `
<div>
  <div>
    <h1>
      HW2CounterList
    </h1>
    <button>
      click me
    </button>
    <ul>
      <li>
        3
      </li>
      <li>
        2
      </li>
      <li>
        1
      </li>
    </ul>
  </div>
</div>
`;

exports[`should add 5 item 1`] = `
<div>
  <div>
    <h1>
      HW2CounterList
    </h1>
    <button>
      click me
    </button>
    <ul>
      <li>
        5
      </li>
      <li>
        4
      </li>
      <li>
        3
      </li>
      <li>
        2
      </li>
      <li>
        1
      </li>
    </ul>
  </div>
</div>
`;
